src/dataflow/src/source/kafka.rs

```diff
@@ -390,7 +403,24 @@ impl KafkaSourceReader {
                     old: self.last_stats.take(),
                     new: Some(stats.clone()),
                 });
-                self.last_stats = Some(stats);
+                self.last_stats = Some(stats.clone());
+            }
+
+            // This temp var is required to satisfy the type checker
```

Comment:
Type annotations like this are very common; not something that requires a specific callout, IMO. You have a few options, too. You could use this syntax, to do it all one one line...

```rust
match serde_json::from_str::<Statistics>(&stats.to_string()) {
    // ...
}
```

...or you can at least omit the bits of the type that type inference can handle:

```rust
let res: Result<Statistics, _> = serde_json::from_str(&stats.to_string());
```